hal/src/main/native/sim/Extensions.cpp

```diff
@@ -0,0 +1,88 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "HAL/Extensions.h"
+
+#include <llvm/SmallString.h>
+#include <llvm/StringRef.h>
+
+#include "HAL/HAL.h"
+
+#if defined(WIN32) || defined(_WIN32)
+#include <windows.h>
+#else
+#include <dlfcn.h>
+#endif
+
+#if defined(WIN32) || defined(_WIN32)
+#define DELIM ';'
+#define HTYPE HMODULE
+#define PREFIX NULL
+#define SUFFIX NULL
+#define DLOPEN(a) LoadLibrary(a)
+#define DLSYM GetProcAddress
+#define DLCLOSE FreeLibrary
+#else
+#define DELIM ':'
+#define HTYPE void*
+#define PREFIX "lib"
+#if defined(__APPLE__)
+#define SUFFIX ".dylib"
+#else
+#define SUFFIX ".so"
+#endif
+#define DLOPEN(a) dlopen(a, RTLD_LAZY)
+#define DLSYM dlsym
+#define DLCLOSE dlclose
+#endif
+
+extern "C" {
+
+int HAL_LoadOneExtension(const char* library) {
+  int rc = 1;  // It is expected and reasonable not to find an extra simulation
+  HTYPE handle = DLOPEN(library);
+  if (!handle && (PREFIX || SUFFIX)) {
+    llvm::SmallString<128> library_name;
```

Comment:
Our style guide uses camel case instead of snake case.

Reply:
Have I missed a pointer to the correct style guide?  This is what I've read:
  https://github.com/wpilibsuite/styleguide/blob/master/cppguide.html#L667
which says 'The names of variables and data members are all lowercase, with
underscores between words'.

Reply:
Oops. Looks like we forgot to update that. We initially chose the Google style guide, then modified it a bit to match our current style. I'll need to find time this weekend to comb through it and remove legacy rules and bring it into alignment with allwpilib's code base. You can use allwpilib as a template for the prevailing style (that is, camel case with an uppercase first letter in C++ and a lowercase first letter in Java for functions). Member variables are `m_likeThis` and normal variables are `likeThis`.